### PR TITLE
Support new representation of target-typed new

### PR DESCRIPTION
**Pull request by sharwell**
Workaround for https://github.com/dotnet/roslyn/issues/59171 in the event the original behavior is not restored.

### CODE REVIEW

**Comment by Youssef1313** [comment]
https://github.com/dotnet/roslyn-analyzers/blob/7c544ae66c44f1705ef583b112def4db52c0179a/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseCancellationTokenThrowIfCancellationRequested.cs#L139

Needs an update as well?

File: src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticDescriptorCreationAnalyzer.cs [hunk 1]
```diff
@@ -393,7 +393,11 @@ static void FreeLocalizableStringsMap(PooledLocalizabeStringsConcurrentDictionar
             [NotNullWhen(returnValue: true)] out IMethodSymbol? creationMethod,
             [NotNullWhen(returnValue: true)] out ImmutableArray<IArgumentOperation> creationArguments)
         {
-            (creationMethod, creationArguments) = fieldInitializer.Value switch
+            var value = fieldInitializer.Value;
+            if (value is IConversionOperation { IsImplicit: true } conversion)
+                value = conversion.Operand;
```
Comment (Youssef1313) [inline src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticDescriptorCreationAnalyzer.cs:398]: Consider using `WalkDownConversion`

**Review by mavasani — APPROVED** [review]

**Review by Youssef1313 — COMMENTED** [review]
Also this one may have the same bug:

https://github.com/dotnet/roslyn-analyzers/blob/7c544ae66c44f1705ef583b112def4db52c0179a/src/NetAnalyzers/Core/Microsoft.NetFramework.Analyzers/DoNotUseInsecureXSLTScriptExecution.cs#L230

This one might need a separate PR though since security analyzers changes go in 2.9.x branch as far as I know.

**Comment by mavasani** [comment]
@sharwell Is this PR ready?

**Comment by mavasani** [comment]
@sharwell The Roslyn issue was closed as by-design. Should we merge in this change?

**Comment by Youssef1313** [comment]
Already fixed in https://github.com/dotnet/roslyn-analyzers/pull/6119. I'm going to close.

**Pull request closed without merging**